packages/ui/__tests__/Card.test.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import { mountAndCheckA11Y } from '@hazelcast/test-helpers'
+
+import { Card, CardProps, Icon } from '../src'
+import { Archive, Database } from 'react-feather'
+import { IconButton } from '../src/IconButton'
+
+describe('Card', () => {
+  const cardTitle = 'Card title'
+  const cardContent = 'Card content'
+
+  const cardLayoutTestData: (CardProps & { headingType?: 'H2' | 'H3' })[] = [
+    {
+      title: cardTitle,
+      headingType: 'H2',
+    },
+    {
+      title: cardTitle,
+      type: 'primary',
+      headingType: 'H2',
+      icon: Database,
+    },
+    {
+      title: cardTitle,
+      type: 'secondary',
+      headingType: 'H3',
+      iconButton: {
+        iconAriaLabel: 'label',
+        icon: Archive,
+      },
+    },
+    {
+      title: cardTitle,
+      type: 'highlighter',
+      headingType: 'H3',
+    },
+    {
+      type: 'primary',
+    },
+    {
+      type: 'secondary',
+    },
+  ]
+
+  it.each(cardLayoutTestData)('Renders correct layout for specific component configurations.', async ({ headingType, ...rest }) => {
+    const wrapper = await mountAndCheckA11Y(<Card {...rest}>{cardContent}</Card>)
+
+    if (rest.title) {
+      expect(wrapper.findDataTest('card-heading').first().getDOMNode().tagName).toBe(headingType)
```

Comment:
You can safely remove `.first()` here:
```suggestion
      expect(wrapper.findDataTest('card-heading').getDOMNode().tagName).toBe(headingType)
```